packages/core/admin/ee/server/seat-enforcement.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+const _ = require('lodash');
+
+const enableUsersToLicenseLimit = async (numberOfUsersToEnable) => {
+  const data = await strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+
+  if (!data || !data.value || data.value.length === 0) return;
+
+  const disabledUsers = JSON.parse(data.value);
+
+  const usersToEnable = _.take(disabledUsers, numberOfUsersToEnable);
+
+  usersToEnable.forEach(async (user) => {
+    const data = await strapi.db.query('admin::user').findOne({
+      where: { id: user.id },
+    });
+
+    if (!data) return;
+
+    await strapi.db.query('admin::user').update({
+      where: { id: user.id },
+      data: { isActive: true },
+    });
```

Comment:
Why don't you just try to update with the second query and handle the result instead of making two queries for each user? :thinking: